### ISSUE TITLE
Warnings about `__FILE__ in eval` after upgrading to Ruby 2.7

### ISSUE THREAD

**Issue by guss77** [njonsson/autoloaded#13]
With a gem that uses autoloaded, after updating to Ruby 2.7, we start getting these warnings:

```
/home/odeda/.gem/ruby/2.7.0/gems/cloudformation-tool-1.3.3/lib/cloud_formation_tool.rb:42: warning: __FILE__ in eval may not return location in binding; use Binding#source_location instead
/home/odeda/.gem/ruby/2.7.0/gems/autoloaded-2.2.1/lib/autoloaded/autoloader.rb:262: warning: in `eval'
/home/odeda/.gem/ruby/2.7.0/gems/cloudformation-tool-1.3.3/lib/cloud_formation_tool/cli.rb:3: warning: __FILE__ in eval may not return location in binding; use Binding#source_location instead
/home/odeda/.gem/ruby/2.7.0/gems/autoloaded-2.2.1/lib/autoloaded/autoloader.rb:262: warning: in `eval'
/home/odeda/.gem/ruby/2.7.0/gems/cloudformation-tool-1.3.3/lib/cloud_formation_tool/cloud_formation.rb:7: warning: __FILE__ in eval may not return location in binding; use Binding#source_location instead
/home/odeda/.gem/ruby/2.7.0/gems/autoloaded-2.2.1/lib/autoloaded/autoloader.rb:262: warning: in `eval'
```

The gem in question - cloudformation-tool - does not call neither `__FILE__` nor `eval` so I think it comes from autoloaded. 

The problem seemed to be introduced by this: https://bugs.ruby-lang.org/issues/14230, and is triggered by `Autoloaded::Autoloader#host_source_location`.

**Comment by njonsson** [comment]
Thanks for the heads-up. Investigating …

**Comment by njonsson** [comment]
Thanks for the contribution!

**Comment by njonsson** [comment]
Resolved in [v2.3.0](https://rubygems.org/gems/autoloaded/versions/2.3.0) and [v1.6.0](https://rubygems.org/gems/autoloaded/versions/1.6.0).
